test/decl/ext/protocol.swift

```diff
@@ -971,7 +971,7 @@ extension BadProto4 { } // okay
 typealias RawRepresentableAlias = RawRepresentable
 extension RawRepresentableAlias { } // okay
 
-extension AnyObject { } // expected-error{{non-nominal type 'AnyObject' cannot be extended}}
+extension AnyObject { } // expected-error{{builtin type 'AnyObject' cannot be extended}}
```

Comment:
'builtin type' is not a very good term either. We should could add a tailored diagnostic for AnyObject

Reply:
In the educational note, it's simply referred to as a "special type," which I think is fine.

Reply:
Sorry, I made a mistake here. It would actually say "typealias". I have added a tailored diagnostic though.